src/main/java/com/griddynamics/cloud/learning/service/ArticleService.java

```diff
@@ -77,4 +91,34 @@ public ArticleService(ArticleRepository repository, ArticleMapper articleMapper)
                 .map(articleMapper::articleToArticleDto)
                 .collect(Collectors.toList()), articles.getPageable(), articles.getTotalElements());
     }
+
+    public ArticleLikeDto likeAnArticle(final ArticleLikeDto likeDto) {
+
+        final String currentUserUsername = getCurrentUserUsername();
+        final Long currentArticleId = likeDto.getArticleId();
+        final Boolean liked = likeDto.getLiked();
+
+        final UserArticle userArticle = StreamSupport.stream(userArticleRepository.findAll().spliterator(), Boolean.FALSE)
+                .filter(ua -> ua.getUser().getUsername().equals(currentUserUsername)
+                        && ua.getArticle().getId().equals(currentArticleId))
+                .findFirst()
+                .orElseGet(() -> createUserArticle(currentUserUsername, currentArticleId, liked));
+
+        userArticle.setLiked(liked);
+
+        userArticleRepository.save(userArticle);
+
+        return likeDto;
+    }
+
+    private UserArticle createUserArticle(final String username, final Long articleId, final Boolean liked) {
```

Comment:
if you would like to call to like this - please, keep userArticleRepository.save(userArticle); within as well